providers/provider-types.go

```diff
@@ -0,0 +1,63 @@
+package providers
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"strings"
+	"sync"
+)
+
+type NodeProvider interface {
```

Comment:
Lets rename it to NodeConfig provider to be clear that it's not actually provisioning the node, but just returning the config. Readme has to be changed accordingly